catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -69,45 +75,59 @@ public CatalogBackupPlugin() {
      * Backs up created metacards to the file system backup.
      *
      * @param input the {@link CreateResponse} to process
-     * @return
+     * @return {@link CreateResponse}
      * @throws PluginExecutionException
      */
     @Override
     public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+
         if (enableBackupPlugin && Requests.isLocal(input.getRequest())) {
-            LOGGER.debug("Performing backup of metacards in CreateResponse.");
+            LOGGER.trace("Performing backup of metacards in CreateResponse.");
 
             if (rootBackupDir == null) {
                 throw new PluginExecutionException("No root backup directory configured.");
             }
 
-            List<String> errors = new ArrayList<String>();
+            EXECUTOR_SERVICE.submit(() -> processBatch(input.getCreatedMetacards()));
```

Comment:
There might only be one metacard to backup. I think I'd have to some period of time or a combination of time period and size. 

They are arriving in batches of 500 now. I'm guessing they are being batched by the IngestCommand. So if anyone other than Ingest was doing it, the batch size could be 1, or everything, or something in between.